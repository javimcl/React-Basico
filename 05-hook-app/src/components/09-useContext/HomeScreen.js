import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    const userContext = useContext(UserContext);
    console.log(userContext);
  return (
  <div>
      <h1>HomeScreen</h1>
      <hr/>

  </div>
  );
};
