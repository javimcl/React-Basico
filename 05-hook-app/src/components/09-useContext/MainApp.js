import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    const user = {
        id: 123,
        name:'Javier',
        email: 'fg@gmail.com'
    }
    return (
        <UserContext.Provider value={user}>
            <AppRouter />
        </UserContext.Provider>

    );
};
