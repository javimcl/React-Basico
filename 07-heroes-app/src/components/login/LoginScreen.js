import React from 'react';

export const LoginScreen = () => {
    return (
        <div>
            <h1>Hola Login screeen</h1>
        </div>
    );
};
