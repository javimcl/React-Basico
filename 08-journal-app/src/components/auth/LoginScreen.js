import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { login } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {

  const dispatch = useDispatch();

  const [formValues, handleInputChange] = useForm({
    email: 'javier',
    password: '1234'
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(login(123,'Javier'))

  }


  return (
    <>
      <h3 className="auth__title">Login</h3>
      <form onSubmit={handleLogin}>
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          autoComplete='off'
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
          value={password}
          onChange={handleInputChange}
        />
        <button className='btn btn-primary btn-block'
          type="submit"
        >
          Login
        </button>

        <div className='auth__social-networks'>
          <p>Login with social metworks</p><div
            className="google-btn"
          >
            <div className="google-icon-wrapper">
              <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className='link'
          to="/auth/register">
          Create new account
        </Link>

      </form>
    </>
  );
};
