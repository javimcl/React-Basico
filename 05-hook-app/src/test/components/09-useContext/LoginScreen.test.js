import { render } from "@testing-library/react";
import { shallow, mount } from 'enzyme';
import '@testing-library/jest-dom';
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";


describe('Pruebas en <LoginScreen/>', () => {

   const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>

            <LoginScreen />
        </UserContext.Provider>

    )

    test('debe de mostrarse correctamecte', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de ejecutar el setUser con el argumento', () => {
      
        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            id: 123,
            name: 'Javier'
        })
        
    });
    

});
