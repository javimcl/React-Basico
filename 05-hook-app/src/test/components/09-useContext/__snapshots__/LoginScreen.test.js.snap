// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe de mostrarse correctamecte 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
